spec/unit/form_builder_spec.rb

```diff
@@ -98,20 +98,20 @@ def build_form(options = {}, form_object = Post.new, &block)
     end
 
     it "should generate a text input" do
-      expect(body).to have_selector("input[type=text][name='post[title]']")
+      expect(body).to have_field("post[title]", type: "text")
     end
 
     it "should generate a textarea" do
-      expect(body).to have_selector("textarea[name='post[body]']")
+      expect(body).to have_css("textarea[name='post[body]']")
     end
 
     it "should only generate the form once" do
-      expect(body).to have_selector("form", count: 1)
+      expect(body).to have_css("form", count: 1)
     end
 
     it "should generate actions" do
-      expect(body).to have_selector("input[type=submit][value='Submit Me']")
-      expect(body).to have_selector("input[type=submit][value='Another Button']")
+      expect(body).to have_css("input[type=submit][value='Submit Me']")
+      expect(body).to have_css("input[type=submit][value='Another Button']")
```

Comment:
Shouldn't this use the `have_button` matcher?

Reply:
The matcher works, I've did the change in a separate commit, because `have_button` does not allow to specify the type, in case of revert